src/notebook/handler.ts

```diff
@@ -47,18 +47,64 @@ import {
   NotebookPanel
 } from './widget';
 
+import {
+  ISignal, Signal, clearSignalData
+} from 'phosphor-signaling';
+
+
+export
+class NotebookContext {
```

Comment:
This is sort of a useless class right now. I use this interface instead of just passing the panel directly so that we have more flexibility in the future. It's definitely just a test case right now to see wbat the needs are.